src/jace/jax/translation_cache.py

```diff
@@ -0,0 +1,290 @@
+        return next_stage
+
+    return _action_wrapper
+
+
+def clear_translation_cache() -> None:
+    """Clear all caches associated to translation."""
+    _TRANSLATION_CACHES.clear()
+
+
+def get_cache(
+    stage: CachingStage,
+) -> TranslationCache:
+    """Returns the cache that is used for `stage`."""
+    # The caches are per stage and not per instance basis
+    tstage = type(stage)
+    if tstage not in _TRANSLATION_CACHES:
+        _TRANSLATION_CACHES[tstage] = TranslationCache(size=_DEF_CACHE_SIZE)
+    return _TRANSLATION_CACHES[tstage]
+
+
+@dataclasses.dataclass(frozen=True)
+class _AbstractCallArgument:
+    """Class to represent one argument to the call in an abstract way.
+
+    It is used as part of the key in the cache.
+    It represents the structure of the argument, i.e. its shape, type and so on, but nots its value.
+    To construct it you should use the `from_value()` class function which interfere the characteristics from a value.
+    """
+
+    shape: tuple[int, ...]
+    dtype: dace.typeclass
+    strides: tuple[int, ...] | None
+    storage: dace.StorageType
+
+    @classmethod
+    def from_value(
+        cls,
+        val: Any,
+    ) -> _AbstractCallArgument:
+        """Construct an `_AbstractCallArgument` from a value.
+
+        Todo:
+            Handle storage location of arrays correctly.
+        """
+        if not util.is_fully_addressable(val):
+            raise NotImplementedError("Distributed arrays are not addressed yet.")
+        if isinstance(val, jax_core.Literal):
+            raise TypeError("Jax Literals are not supported as cache keys.")
+
+        if util.is_array(val):
+            if util.is_jax_array(val):
+                val = val.__array__()  # Passing `copy=False` leads to error in NumPy.
+            shape = val.shape
+            dtype = util.translate_dtype(val.dtype)
+            strides = getattr(val, "strides", None)
+            # Is `CPU_Heap` always okay? There would also be `CPU_Pinned`.
+            storage = (
+                dace.StorageType.GPU_Global if util.is_on_device(val) else dace.StorageType.CPU_Heap
+            )
+
+            return cls(shape=shape, dtype=dtype, strides=strides, storage=storage)
+
+        if util.is_scalar(val):
+            shape = ()
+            dtype = util.translate_dtype(type(val))
+            strides = None
+            # Scalar arguments are always on the CPU and never on the GPU.
+            storage = dace.StorageType.CPU_Heap
+
+            return cls(shape=shape, dtype=dtype, strides=strides, storage=storage)
+
+        raise TypeError(f"Can not make 'an abstract description from '{type(val).__name__}'.")
+
+
+#: This type is the abstract description of a function call.
+#:  It is part of the key used in the cache.
+CallArgsDescription: TypeAlias = tuple[
+    _AbstractCallArgument | Hashable | tuple[str, _AbstractCallArgument | Hashable],
+    ...,
+]
+
+
+@dataclasses.dataclass(frozen=True)
+class CachedCallDescription:
```

Comment:
This object represents the information of a call to a specific Stage method (or _action_, as it is sometimes named in this module) with a specific set of arguments. I think the name is misleading because it doesn't mention that it represents a call to a Stage transformation method and instead it mentions _caching_, which is completely unrelated to the content of the object (using this object for caching is just  one possible use).

I would consider using other names like `ActionCallDescription`  or `StageTransformationCall` or something along this line.

Reply:
I used `StageTransformationDescription`.